python/src/lib.rs

```diff
@@ -327,6 +327,7 @@ impl RawDeltaTable {
     /// This will acquire the internal lock since it is a mutating operation!
     pub fn load_version(&self, py: Python, version: i64) -> PyResult<()> {
         py.allow_threads(|| {
+            #[allow(clippy::await_holding_lock)]
```

Comment:
Clippy is complaining about this -- it might be a real issue, but I didn't want to mess with it is as part of this PR. This PR makes the situation better in my opinion, as now the clippy warning is explicit in the code